libs/vkd3d/command.c

```diff
@@ -7353,19 +7281,8 @@ static void STDMETHODCALLTYPE d3d12_command_list_ResourceBarrier(d3d12_command_l
                      * quite weird. We cannot perform UNDEFINED transitions here, even if that is what makes sense.
                      * UNDEFINED transitions are deferred to their required "activation" command, which is a full-subresource
                      * command that writes every pixel. We detect those cases and perform a transition away from UNDEFINED. */
-                    if (before && d3d12_resource_is_texture(before))
-                    {
-                        alias_src_access = vk_access_flags_all_possible_for_image(list->device,
-                                before->desc.Flags,
-                                list->vk_queue_flags,
-                                false);
-                    }
-                    else
-                    {
-                        alias_src_access = vk_access_flags_all_possible_for_buffer(list->device,
-                                list->vk_queue_flags,
-                                false);
-                    }
+                    if (before)
```

Comment:
I think this is a bug if resource is a buffer. This should be unconditionally added I think.

Reply:
Fixed.